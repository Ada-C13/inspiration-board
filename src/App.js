import React from "react";
import "./App.css";
import Board from "./components/Board";

const App = () => {
  return (
    <section>
      <header className="header">
        <h1 className="header__h1">
          <span className="header__text">Inspiration Board</span>
        </h1>
      </header>
      <Board
        url="https://inspiration-board.herokuapp.com/boards/"
        boardName={`Pinreact`} //let's pick a name
      />
    </section>
  );
};

export default App;

// ------ PROPS -------
//formSubmitCallback
//allBoardsState

// ------- CREATING NEXT ID ------

//----- GET API ----------------
//const App = () => {
// const [studentList, setStudentList] = useState([]);
//;

// useEffect(() => {
//   axios.get(API_URL_BASE)
//     .then((response) => {
//       const apiStudentList = response.data;
//       setStudentList(apiStudentList);
//     })
//     .catch((error) => {
//       setErrorMessage(error.message);
//       console.log(error.message);
//     });
// }, []);

//-----RETURN THE ERROR MESSAGE------
//<div className="App">
//{ errorMessage ? <div><h2 className="error-msg">{errorMessage}</h2></div> : '' }
