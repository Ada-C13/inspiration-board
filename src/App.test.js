import React from 'react';
import { render, cleanup, getByText } from '@testing-library/react'
import App from './App'

describe('App', () => {
  test('that it matches the existing snapshot', () => {
    // Arrange-Act
    const { asFragment } = render(
      <App />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });

    test('An App can be rendered', () => {
      const renderSampleApp = () => render(<App />);
      const { getByText } = renderSampleApp();
      const BoardElement = getByText(/Inspiration Board/i);
      expect(BoardElement).toBeInTheDocument;
  });
});

// describe('App', () => {
//   test('renders Inpiration Board', () => {
//     const { getByText } = render(<App />);
//     const linkElement = getByText(/Inspiration Board/i);
//     expect(linkElement).toBeInTheDocument();
//     cleanup();
//   });
// });

//   it('renders without crashing', () => {
//     console.log('testing would be nice! :)');
//   });

// });

// describe('App', () => {
//   afterEach(cleanup);
//   const renderSampleApp = () => render(<App />);

// });


// src/components/test/NewStudentForm.test.js

