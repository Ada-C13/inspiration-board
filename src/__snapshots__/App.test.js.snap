// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <section>
    <header
      class="header"
    >
      <h1
        class="header__h1"
      >
        <span
          class="header__text"
        >
          Inspiration Board
        </span>
      </h1>
    </header>
    <div
      class="board"
      title="Sharon-Olga"
    >
      <form
        class="new-card-form"
      >
        <section
          class="new-card-form__form"
        >
          <header
            class="new-card-form__header"
          >
            Add a new card
          </header>
          <div
            class="new-card-form__form-label"
          >
            Type an inspiration text here
          </div>
          <input
            class="new-card-form__form-textarea"
            name="text"
            value=""
          />
          <div
            class="new-card-form__form-label"
          >
            Add an Emoji
          </div>
          <select
            class="new-card-form__form-select"
            name="emoji"
          >
            <option />
            <option>
              😍
            </option>
            <option>
              🍺
            </option>
            <option>
              👏
            </option>
            <option>
              💖
            </option>
            <option>
              😻
            </option>
            <option>
              🐶
            </option>
          </select>
          <button
            class="new-card-form__form-button"
            type="submit"
          >
            Submit
          </button>
        </section>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
